src/tiled/scriptdialog.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+ * scriptdialog.cpp
+ * Copyright 2020, David Konsumer <konsumer@jetboystudio.com>
+ * Copyright 2020, Thorbjørn Lindeijer <bjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "scriptdialog.h"
+#include "scriptmanager.h"
+#include <QProcessEnvironment>
+#include <QJSEngine>
+#include <string.h>
+#include <typeinfo>
+#include <memory>
+
+namespace Tiled {
+
+ScriptDialog::ScriptDialog(): ScriptDialog(QString(), 450, 450)
+{
+    
+}
+ScriptDialog::ScriptDialog(const QString &title, const int width=450, const int height=450)
+ : QDialog(MainWindow::maybeInstance())
+{
+    if (title.isEmpty()){
+        setTitle(QString::fromUtf8("Script"));
```

Comment:
That should be translatable:
```suggestion
        setTitle(tr("Script"));
```